routing/additionaledge.go

```diff
@@ -0,0 +1,58 @@
+package routing
+
+import (
+	"errors"
+
+	"github.com/lightningnetwork/lnd/channeldb"
+	"github.com/lightningnetwork/lnd/lnwire"
+	"github.com/lightningnetwork/lnd/routing/route"
+)
+
+var (
+	// ErrNoPayLoadSizeFunc is returned when no payload size function is
+	// definied.
+	ErrNoPayLoadSizeFunc = errors.New("no payloadSizeFunc definied for " +
+		"additional edge")
+)
+
+// customEdgeSizeFunc is a size function that is used to allow edges to specify
+// a custom hop size for the forwarding variables provided.
+type customEdgeSizeFunc func(amount lnwire.MilliSatoshi, expiry uint32,
+	legacy bool, channelID uint64) uint64
+
+// hopPayloadSizeFunc calculates the size of the payload for a clear hop of
+// a route (non-blinded). This size function must only be used for intermediate
+// hops. The exit hop payload has to be treated differently.
+func hopPayloadSizeFunc(amount lnwire.MilliSatoshi, expiry uint32,
+	legacy bool, channelID uint64) uint64 {
+
+	hop := route.Hop{
+		AmtToForward:     amount,
+		OutgoingTimeLock: expiry,
+		LegacyPayload:    legacy,
+	}
+
+	return hop.PayloadSize(channelID)
+}
+
+// AdditionalEdge encapsulates the policy and the payload size function for
+// additional edges which are amended to a path. These can either be blinded
+// hops or additional routing hops/hints (unannounced channels).
+type AdditionalEdge struct {
+	policy          *channeldb.CachedEdgePolicy
+	payloadSizeFunc customEdgeSizeFunc
+}
+
+// EdgePolicy return the policy of the AdditionalEdge.
+func (edge *AdditionalEdge) edgePolicy() *channeldb.CachedEdgePolicy {
```

Comment:
should we export those functions, although we do not use them currently from the outside? We need to export AdditionalEdge because its used in the lnrpc package. Moreover the RouteRequests has the `RouteHints` as an element. 
I was trying to move the route hint calculation from the lnrpc package into the routing package (where its already done for the blinded path) but as long as we export the RouteHints in the RouteRequest seems to be its not a big win?

Reply:
I think we should export it if it's used by the `lnrpc`.

Reply:
Agree Exported all functions, also the `PayloadSize` one.